pegen/pegen.h

```diff
@@ -97,6 +98,22 @@ void *CONSTRUCTOR(Parser *p, ...);
 #define EXTRA_EXPR(head, tail) head->lineno, head->col_offset, tail->end_lineno, tail->end_col_offset, p->arena
 #define EXTRA start_lineno, start_col_offset, end_lineno, end_col_offset, p->arena
 
+jmp_buf error_marker;
+
+#define PROPAGATE_ERROR(errno) \
+    if (!PyErr_Occurred()) { \
+        PyErr_Format(PyExc_RuntimeError, "Unknown error occured when parsing"); \
+    } \
+    longjmp(error_marker, errno); \
+    Py_UNREACHABLE();
+
+static inline void CHECK_CALL(void* result) {
```

Comment:
This macro will be used in the helpers to check results (see https://github.com/gvanrossum/pegen/pull/130#issuecomment-566775430)